custom/enikshay/private_sector_datamigration/factory.py

```diff
@@ -0,0 +1,299 @@
+            kwargs['attrs']['update']['dob'] = self.beneficiary.dob.date()
+            kwargs['attrs']['update']['dob_known'] = 'yes'
+        else:
+            if self.beneficiary.age_entered is not None:
+                kwargs['attrs']['update']['dob'] = date(date.today().year - self.beneficiary.age_entered, 7, 1),
+            else:
+                kwargs['attrs']['update']['dob'] = ''
+            kwargs['attrs']['update']['dob_known'] = 'no'
+
+        if self.beneficiary.sex is not None:
+            kwargs['attrs']['update']['sex'] = self.beneficiary.sex
+
+        if self.beneficiary.has_aadhaar_number:
+            kwargs['attrs']['update']['aadhaar_number'] = self.beneficiary.identificationNumber
+        else:
+            kwargs['attrs']['update']['other_id_number'] = self.beneficiary.identificationNumber
+            kwargs['attrs']['update']['other_id_type'] = self.beneficiary.other_id_type
+
+        if self._episode:
+            kwargs['attrs']['update']['hiv_status'] = self._episode.hiv_status
+            kwargs['attrs']['update']['current_patient_type_choice'] = self._episode.current_patient_type_choice
+
+        agency = (
+            self._episode.treating_provider or self.beneficiary.referred_provider
+            if self._episode else self.beneficiary.referred_provider
+        )
+        assert agency is not None
+
+        kwargs['attrs']['owner_id'] = SQLLocation.active_objects.get(
+            domain=self.domain,
+            site_code=agency.nikshayId,
+        ).location_id
+
+        return CaseStructure(**kwargs)
+
+    def get_occurrence_case_structure(self, person_structure):
+        kwargs = {
+            'attrs': {
+                'case_type': OCCURRENCE_CASE_TYPE,
+                'close': False,
+                'create': True,
+                'owner_id': '-',
+                'update': {
+                    'current_episode_type': self.beneficiary.current_episode_type,
+                    'name': 'Occurrence #1',
+                    'occurrence_id': get_human_friendly_id(),
+
+                    'migration_created_case': 'true',
+                    'migration_created_from_record': self.beneficiary.caseId,
+                }
+            },
+            'indices': [CaseIndex(
+                person_structure,
+                identifier='host',
+                relationship=CASE_INDEX_EXTENSION,
+                related_type=PERSON_CASE_TYPE,
+            )],
+        }
+        return CaseStructure(**kwargs)
+
+    def get_episode_case_structure(self, occurrence_structure):
+        kwargs = {
+            'attrs': {
+                'case_type': EPISODE_CASE_TYPE,
+                'close': False,
+                'create': True,
+                'owner_id': '-',
+                'update': {
+                    'adherence_schedule_id': 'schedule_mwf',
+                    'date_of_mo_signature': self.beneficiary.dateOfRegn.date(),
+                    'dots_99_enabled': 'false',
```

Comment:
I think there might be some patients already enrolled in 99DOTS in their system... Need to think how we're going to link those up.